drivers/SmartThings/zigbee-button/src/frient/init.lua

```diff
@@ -34,18 +43,121 @@ local battery_table = {
+  ledColor = {
+    clusterId = OnOff.ID,
+    attributeId = BUTTON_LED_COLOR,
+    dataType = data_types.Enum8,
+    mfg_code =  DEVELCO_MANUFACTURER_CODE,
+    endpoint = 0x20,
+    frame_ctrl = 0x0C
+  },
+  buttonDelay = {
+    clusterId = OnOff.ID,
+    attributeId = BUTTON_PRESS_DELAY,
+    dataType = data_types.Uint16,
+    mfg_code =  DEVELCO_MANUFACTURER_CODE,
+    endpoint = 0x20,
+    frame_ctrl = 0x0C
+  },
+  panicButton = {
+    clusterId = BasicInput.ID,
+    attributeId = PANIC_BUTTON,
+    dataType = data_types.Uint16,
+    mfg_code = DEVELCO_MANUFACTURER_CODE,
+    endpoint = 0x20,
+    frame_ctrl = 0x04
+  },
+  buttonAlarmDelay = {
+    clusterId = IASZone.ID,
+    attributeId = 0x8002,
+    dataType = data_types.Uint16,
+    mfg_code = DEVELCO_MANUFACTURER_CODE,
+    endpoint = 0x23,
+    frame_ctrl = 0x04
+  },
+  buttonCancelDelay = {
+    clusterId = IASZone.ID,
+    attributeId = 0x8003,
+    dataType= data_types.Uint16,
+    mfg_code = DEVELCO_MANUFACTURER_CODE,
+    endpoint = 0x23,
+    frame_ctrl = 0x04
+  },
+  autoCancel = {
+    clusterId = IASZone.ID,
+    attributeId = 0x8004,
+    dataType = data_types.Uint16,
+    mfg_code = DEVELCO_MANUFACTURER_CODE,
+    endpoint = 0x23,
+    frame_ctrl = 0x04
+  },
+  alarmBehavior = {
+    clusterId = IASZone.ID,
+    attributeId = 0x8005,
+    dataType = data_types.Enum8,
+    mfg_code = DEVELCO_MANUFACTURER_CODE,
+    endpoint = 0x23,
+    frame_ctrl = 0x04
+  },
+}
+
+local function generate_event_from_zone_status(driver, device, zone_status, zigbee_message)
+  if device:supports_capability(panicAlarm) then
+    if zone_status:is_alarm2_set() then
+      device:emit_event(panicAlarm.panicAlarm.panic({state_change = true}))
+    else
+      device:emit_event(panicAlarm.panicAlarm.clear({state_change = true}))
+    end
+  end
+end
+
+local function configure_ias_zone_settings(driver, device)
+  device:send(cluster_base.write_manufacturer_specific_attribute(device, IASZone.ID, 0x8002, DEVELCO_MANUFACTURER_CODE, data_types.Uint16, 2000):to_endpoint(0x23))
+  device:send(cluster_base.write_manufacturer_specific_attribute(device, IASZone.ID, 0x8003, DEVELCO_MANUFACTURER_CODE, data_types.Uint16, 2000):to_endpoint(0x23))
+  device:send(cluster_base.write_manufacturer_specific_attribute(device, IASZone.ID, 0x8004, DEVELCO_MANUFACTURER_CODE, data_types.Uint16, 10):to_endpoint(0x23))
+  device:send(cluster_base.write_manufacturer_specific_attribute(device, IASZone.ID, 0x8005, DEVELCO_MANUFACTURER_CODE, data_types.Enum8,  0):to_endpoint(0x23))
+end
+
+local function present_value_attr_handler(driver, device, value, zb_rx)
   if value.value == true then
-    event = capabilities.button.button.pushed(additional_fields)
-    device:emit_event(event)
+    device:emit_event(capabilities.button.button.pushed({state_change = true}))
   end
 end
 
-local function init_handler(self, device, event, args)
+local function ias_zone_status_attr_handler(driver, device, zone_status, zb_rx)
```

Comment:
You could just register `generate_event_from_zone_status` as the handler for the IAS Zone Status attribute rather than including this extra passthrough function.